### PR TITLE
Implement mod resolving for FTB

### LINKED ISSUE #955: FTB 'url's parent does not contain 'url'
### Operating System

Linux

### Version of PolyMC

PolyMC 1.3.2 and PolyMC 1.4.0

### Description of bug

I was trying to download FTB University 1.16 with my friend who is on windows and it shows us the error: 

Could not understand pack manifest:
'url's parent does not contain 'url'

### Steps to reproduce

Trying to download a FTB modpack shows the error.

### Suspected cause

_No response_

### This issue is unique

- [x] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Scrumplex**
very ugly, please don't judge

### CODE REVIEW

**Comment by Scrumplex** [comment]
One problem right now: This doesn't handle other blocked assets. 
Take this as an example: https://api.modpacks.ch/public/modpack/95/2132

One of the last entries is a resource pack that would need resolving. `Flame::FileResolvingTask` only handles mods.

**Comment by Scrumplex** [comment]
I enabled access to the branch for maintainers. Feel free to work on this if you want

**Comment by DioEgizio** [comment]
Flame::FileResolvingTask only handles mods.

That's not true

**Comment by DioEgizio** [comment]
image

i don't think this is working correctly, lol

**Comment by DioEgizio** [comment]
oooh i think it's resolving ids but not downloading mods from there

**Comment by Scrumplex** [comment]
Logs or it didn't happen

**Comment by DioEgizio** [comment]
There's literally nothing in logs related to this, just that it sends the stuff to resolve IDs, but download urls are never set to be downloaded

File: launcher/ui/pages/modplatform/ftb/FtbPage.cpp [hunk 1]
```diff
@@ -126,7 +127,7 @@ void FtbPage::suggestCurrent()
         return;
     }
 
-    dialog->setSuggestedPack(selected.name + " " + selectedVersion, new ModpacksCH::PackInstallTask(selected, selectedVersion));
+    dialog->setSuggestedPack(selected.name + " " + selectedVersion, new ModpacksCH::PackInstallTask(selected, selectedVersion, this));
```
Comment (Scrumplex) [inline launcher/ui/pages/modplatform/ftb/FtbPage.cpp:130]: this seems wrong?

Comment (jamierocks) [reply launcher/ui/pages/modplatform/ftb/FtbPage.cpp:130]: Sort of. Previously (when there was a seperation of api and application), you wouldn't have been able to do this.

It's why there is a `UserInteractionSupport` class for the ATLauncher support.

**Comment by DioEgizio** [comment]
seems to work now!

**Comment by DioEgizio** [comment]
ig we can close this for now

**Comment by DioEgizio** [comment]
Ah well, I guess this will have to be reopened

**Comment by Scrumplex** [comment]
Does anyone want to take over this? Seems like this is relevant again

**Comment by adamgoose** [comment]
I have been inconsistently running into this "api breakage" today as well. It appears as though "unknown" mods are returning without at URL.

**Comment by flowln** [comment]
@Scrumplex you need to add your sign-offs :o
see if there's anything else you want to do here, by me this is ready for review i think

**Comment by DioEgizio** [comment]
Epic sign off fail+ you yeeted flow's work away 😭

**Comment by Scrumplex** [comment]
OMG, I literally did not notice. BRUH. @flowln can you force push over my changes again?

**Comment by Scrumplex** [comment]
nvm. did it myself already

**Comment by Scrumplex** [comment]
There we go!

**Review by Scrumplex — COMMENTED** [review]
(My approval for @flowln's changes)

**Review by LennyMcLennington — APPROVED** [review]
I didn't notice anything that would be a problem. Compiled and tested it and it worked fine as far as I can tell.

**Review by flowln — APPROVED** [review]
those Scrumplex guy and flowln girl are super good at what they do :clap: :clap:

**Comment by Scrumplex** [comment]
I fucking despise that Scrumplex guy. Always with his "ooga booga sign-off your commits" and "oh no there is no newline at the end of the file, we can't accept this"

**Comment by DioEgizio** [comment]
lol

**Review by timoreo22 — APPROVED** [review]

**Comment by DioEgizio** [comment]
3 approvals 👀

**Pull request merged**